ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/SendTransactionHandler.java

```diff
@@ -81,6 +84,16 @@ public void handle(final RoutingContext context, final JsonRpcRequest request) {
       return;
     }
 
+    final HexStringComparator comparator = new HexStringComparator();
+    if (comparator.compare(transactionSigner.get().getAddress(), transaction.sender()) != 0) {
+      LOG.info(
+          "Ethereum address derived from identifier ({}) is incorrect value ({})",
+          transaction.sender(),
+          transactionSigner.get().getAddress());
+      context.fail(INTERNAL_SERVER_ERROR.code(), new JsonRpcException(TX_SENDER_NOT_AUTHORIZED));
```

Comment:
Is there a test somewhere for error case TX_SENDER_NOT_AUTHORIZED?

Reply:
we can't hit it atm - not until the protections are removed from Signers repo :(